fw/tls.c

```diff
@@ -820,6 +822,22 @@ tfw_tls_sni(TlsCtx *ctx, const unsigned char *data, size_t len)
 				 PR_TFW_STR(&srv_name));
 			return -ENOENT;
 		}
+
+		/* Save server name for later validation with authority from requests */
+		if (likely(len <= sizeof(ctx->sni_buf))) {
+			ctx->sni_len = len;
+			ctx->sni = ctx->sni_buf;
+			memcpy_fast(ctx->sni_buf, data, len);
+		} else {
+			ctx->sni = kmalloc(len, GFP_KERNEL | GFP_NOWAIT);
```

Comment:
This code is called from softirq, which can not sleep
```suggestion
			ctx->sni = kmalloc(len, GFP_ATOMIC);
```

Reply:
Nobody frees the `sni`

Reply:
It is freed from `tfw_tls_conn_dtor()`